sandbox/reflection/struct.h

```diff
@@ -0,0 +1,192 @@
+#ifndef BRICKS_REFLECTION_STRUCT_H
+#define BRICKS_REFLECTION_STRUCT_H
+
+#include <type_traits>
+
```

Comment:
Perhaps it's best to put the `#include <memory>` here, so that it passes `make check` too.